bin/btmon.py

```diff
@@ -3937,7 +3980,61 @@ def _handle_urlopen_error(self, e, url, payload):
                         '\n  api_key: ' + self.api_key,
                         '\n  system_id: ' + self.system_id,
                         '\n  data:  ' + payload]))
-            
+
+class InfluxDBProcessor(UploadProcessor):
+    def __init__(self, host, port, username, password, database, measurement, map_str, period, timeout):
+        super(InfluxDBProcessor, self).__init__()
+        self.host = host
+        self.port = port
+        self.username = username
+        self.password = password
+        self.database = database
+        self.measurement = measurement
+        self.map_str = map_str
+        self.process_period = int(period)
+        self.timeout = int(timeout)
+        self.map = dict()
+
+        infmsg('InfluxDB: upload period: %d' % self.process_period)
+        infmsg('InfluxDB: host: %s' % self.host)
+        infmsg('InfluxDB: port: %s' % self.port)
+        infmsg('InfluxDB: username: %s' % self.username)
+        infmsg('InfluxDB: map: %s' % self.map_str)
+
+    def setup(self):
+        self.map = pairs2dict(self.map_str)
+
+    def process_calculated(self, packets):
+        sensors = dict()
+        readings = dict()
+        series = []
+        for p in packets:
+            dev_serial = obfuscate_serial(p['serial'])
+            for idx, c in enumerate(PACKET_FORMAT.channels(FILTER_DB_SCHEMA_COUNTERS)):
+                key = mklabel(p['serial'], c)
+                if key in self.map:
+                    tpl = self.map[key]
+                    dev_tag = tpl['tag']
```

Comment:
@chicago6061: i don't think this works. `pairs2dict` turns `foo,bar,baz,bliffl` into `{foo: 'bar', baz: 'bliffl'}` so there'd be no dict with a 'tag' property.